include/git2/repository.h

```diff
@@ -746,6 +747,15 @@ GIT_EXTERN(const char *) git_repository_get_namespace(git_repository *repo);
 GIT_EXTERN(int) git_repository_is_shallow(git_repository *repo);
 
 /**
+ * Determine the shallow roots of the repository
+ *
+ * @param out An array of shallow oids. Can be NULL.
```

Comment:
Since I did the extraction in two pass, it behaved like `git_repository_is_shallow()` (which used it internally). But it doesn't make sense anymore. I'll make it mandatory.